### DISCUSSION TITLE
amazing

### DISCUSSION THREAD

**Discussion by NFHorizon** [lsg2020/skynet_ts#4] [General]
I love deno and skynet deeply, and you bring a project combine them two!

**Comment by lsg2020** [comment]
thanks, welcome to give us advice

**Comment by bijiaoshenqi** [comment]
你改了一些deno和v8的方法，从private改到了public，加了一些ex，才让skynet可以，打印error信息，js传function指针到C。
我不太喜欢改官方的api这种方式。 我尝试不改这些private，直接用deno官方的包，集成的很蛋疼，单纯的service能够起来，但接收消息会莫名其妙的爆栈。且v8引擎把握不住，要注意的东西有点太多了。另外rust，这是我有史以来写的最别扭的语言，一眼难尽。陆陆续续写了一个月，放弃。
最后我找到了duktape，和lua一样有value stack，api也和Lua类似，纯C的，直接仿照snlua就可以写snjs。唯一的缺点是es5，需要自己折腾es6的东西。
虽然关键的skynet.ts需要仿照skynet.lua重新翻译，但十分感谢你写的周边ts，比如lua_seri.ts，节省了不少翻译的时间。
Best appreciate.
